### DISCUSSION TITLE
TSML  v3.11.3 and 12 Step Meeting PDF Generator v0.3.0

### DISCUSSION THREAD

**Discussion by Grateful-Viv** [code4recovery/12-step-meeting-list#549] [Q&A] [answered]
I see there is both an update to both these plugins; TSML (Version 3.11.3) and the Chris Toews  12 Step Meeting PDF Generator (Version: 0.3.0). 

I also see there is a fix in the plugin which generated an error using the external PDF plugin. See #548 Egads! Looking at the View Version Details for 0.3.0, all it references is "fixed bug, thanks to Josh Reisner".

I will test this all out on the staging site, but before I do, can someone please walk me through what was changed? I finally got the TSML plugin to work for our needs (lots of CSS customization) and kinda forced the PDF generator plugin to function okay for us which we use to publish our in-person meetings.

Thank you.

**Comment by Grateful-Viv** [comment, marked as answer]
I did install the updates to the plug-ins on my staging site and all I can see different was pdf which doesn’t currently work because it doesn’t recognize JSON feed. Was anything else done with the update?

**Reply by tech2serve** [reply]
Please [read this thread](https://github.com/code4recovery/12-step-meeting-list/issues/544) and check if your staging site has a valid cert. If it does, and our PDF service still doesn't recognize your JSON feed, please provide more details (errors, screen caps, etc) and we'll try to figure it out. 

Other than transitioning to the PDF service, v3.12 included some code refactoring and cleanup. No other features were introduced or changed. Each release includes some notes in the [readme.txt changelog](https://github.com/code4recovery/12-step-meeting-list/blob/master/readme.txt). 

I also intend to start making announcements here in the Discussions area.

**Reply by Grateful-Viv** [reply]
Thanks for your reply @tech2serve 

Today I checked the PDF generator and got this message: Data is protected. If you are using 12 Step Meeting List, consider setting data sharing to open.

What should the data protection be set at? Should sharing be restricted or open? I have it set to restricted and thought that is what we are supposed to do.

**Reply by tech2serve** [reply]
To use the PDF service, you will need to have it set to open.

**Reply by joshreisner** [reply]
The other option, which is more complicated, would be to create an "Authorized App" for the PDF service, or use the one you created for Meeting Guide.

This unfortunately means the button won't work though. You'll want to copy your key URL and paste it in a pdf.code4recovery.org.
